data/block/block.go

```diff
@@ -116,6 +117,28 @@ func (h *Header) GetMiniBlockHeadersWithDst(destId uint32) map[string]uint32 {
 	return hashDst
 }
 
+// GetOrderedCrossMiniblocksWithDst gets all cross miniblocks with the given destination shard ID, ordered in a
+// chronological way, taking into consideration the round in which they were created/executed in the sender shard
+func (h *Header) GetOrderedCrossMiniblocksWithDst(destId uint32) []*data.MiniBlockInfo {
+	miniBlocks := make([]*data.MiniBlockInfo, 0)
+
+	for _, mb := range h.MiniBlockHeaders {
+		if mb.ReceiverShardID == destId && mb.SenderShardID != destId {
+			miniBlocks = append(miniBlocks, &data.MiniBlockInfo{
+				Hash:          mb.Hash,
+				SenderShardID: mb.SenderShardID,
+				Round:         h.Round,
+			})
+		}
+	}
+
+	sort.Slice(miniBlocks, func(i, j int) bool {
```

Comment:
sort not required. All miniblocks contained by this header will have the exact same round.